pkg/kubelet/container/types.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"errors"
+	"io"
+	"time"
+
+	"github.com/fsouza/go-dockerclient"
+)
+
+var (
+	ErrNoSuchImage = errors.New("no such image")
+)
+
+// Container represents a container.
+type Container struct {
+	ID              string            `json:"id"`
+	Name            string            `json:"name,omitempty"`
```

Comment:
Do we need literally everything docker exposes, or can we slim it down and correct historical mistakes?  Like, do we need 2 name fields?